powersimdata/input/check.py

```diff
@@ -0,0 +1,200 @@
+import networkx as nx
+
+from powersimdata.input.grid import Grid
+
+
+def check_grid(grid):
+    """Check whether an object is an internally-consistent Grid object.
+
+    :param powersimdata.input.grid.Grid grid: grid or grid-like object to check.
+    :raises TypeError: if ``grid`` is not a Grid object.
+    """
+    error_messages = []
+    if not isinstance(grid, Grid):
+        error_messages.append("grid must be a Grid object")
+    _check_attributes(grid, error_messages)
+    _check_for_islanded_buses(grid, error_messages)
+    _check_for_undescribed_buses(grid, error_messages)
+    _check_bus_against_bus2sub(grid, error_messages)
+    _check_ac_interconnects(grid, error_messages)
+    _check_transformer_substations(grid, error_messages)
+    _check_line_voltages(grid, error_messages)
+    _check_plant_against_gencost(grid, error_messages)
+    _check_connected_components(grid, error_messages)
+    if len(error_messages) > 0:
+        raise ValueError(f"Problem(s) found with grid: {' '.join(error_messages)}")
```

Comment:
This is nice! Would it be better to use `'\n'.join(error_messages)` instead?

Reply:
Is there a convention for long error messages, whether they should be line wrapped or not? I am agnostic.

Reply:
Good call. I don't know either. If there is a convention, then we should follow. I'm simply thinking of reading convenience.

Reply:
`\n` won't work in an f-string

Reply:
I see. A work around could be 
![image](https://user-images.githubusercontent.com/52716585/120864960-b8011180-c552-11eb-8e46-e102a426b15f.png)

Reply:
Done.